frontend/src/model/UserProfile.ts

```diff
@@ -0,0 +1,68 @@
+import LiveJsonFile from "./LiveJsonFile";
+import LiveDirectory, {PathData} from "./LiveDirectory";
+
+type PathType = 'profile' | '404' | 'loading';
+
+class UserProfile {
+    dir: LiveDirectory;
+    profileJson: LiveJsonFile;
+
+    constructor(dir: LiveDirectory) {
+        this.dir = dir;
+        this.profileJson = this.dir.getJsonFile('profile.json');
+    }
+
+    getAttribute(key: string, defaultValue: string) : string {
+        return this.profileJson.getAttribute(key, defaultValue)
+    }
+
+    setAttribute(key: string, value: any) {
+        this.profileJson.getAttribute(key, value)
+    }
+
+    getProfileFullName() : string {
+        let name:string = this.profileJson.getAttribute("name", "");
+        let surname:string = this.profileJson.getAttribute("surname", "");
+        if (this.profileJson.loading) {
+            return "(Loading...)";
+        }
+        else if (name !== "" && surname !== "") {
+            return name + " " + surname;
+        }
+        else if  (name === "" && surname !== "") {
+            return surname;
+        }
+        else if (name !== "" && surname === "") {
+            return name;
+        }
+        else {
+            return "";
+        }
+    }
+
+    getPathType(path: string): PathType {
+
+        if (path.startsWith('/')) {
+            path = path.substring(1);
+        }
+
+        const pathData: PathData | undefined = this.dir.getPath(path, false)
```

Comment:
@keenon Here, I am doing getPath instead of getCachedPath because at this point, if you go to the profile directly, the getPath to this path has never been called before, so it has never been cached. I think the function getPath is internally caching the first time it is calle, so if it is being called in consecutive renders, it should just get the cached path the following renders. Is this correct?